projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+	delay_ms(5);
+
+	flushBuff(pBuff, sizeof(pBuff));
+	eSensorResult = pGasSensor->ReadTIAGain(pBuff);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"RTIA = %s ohm%s",rtia[pBuff[0]], _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	flushBuff(pBuff, sizeof(pBuff));
+	eSensorResult = pGasSensor->ReadRload(pBuff);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Rload = %s ohm%s",rload[pBuff[0]], _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	eSensorResult = pGasSensor->ReadSensorBias(&sigVal);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Vbias = %d mV%s",sigVal, _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	eSensorResult = pGasSensor->ReadSensorSensitivity(&unsiglVal);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Sensitivity = %.2f nA/ppm%s", (unsiglVal / 100.0), _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	eSensorResult = pGasSensor->ReadMeasurementTime(&unsigVal);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	sprintf((char*)TXbuff,"Measurement Time = %d msec%s", unsigVal, _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);
+
+	// === NOT USED IN THIS VERSION! ===
+	/*flushBuff(pBuff, sizeof(pBuff));
+	eSensorResult = pGasSensor->ReadTempCompCfg(pBuff);
+	if(eSensorResult != SENSOR_ERROR_NONE) return eSensorResult;
+	flushBuff(TXbuff, sizeof(TXbuff));
+	if (pBuff[0] == 0) sprintf((char*)TXbuff,"Temp Comp = disabled%s", _EOS);
+	if (pBuff[0] == 1) sprintf((char*)TXbuff,"Temp Comp = enabled%s", _EOS);
+	UART_TX((const char*)TXbuff);
+	delay_ms(5);*/
+
+	eSensorResult = pGasSensor->close();
+	delay_ms(5);
+
+	return eSensorResult;
+}
+
+/*!
+ * @brief      Reads averaged ppb value from gas sensor
+ *
+ * @details    Reads sensor data utilizing custom gas sensor I2C library and sends this data over UART.
+ */
+int32_t SensorReadoutPPB(uint8_t sensor_address)
+{
+	int32_t PPB_Gas_Reading = 0;
+	eSensorResult = pGasSensor->openWithAddr(sensor_address);
+	eSensorResult = pGasSensor->ReadDataPPB(&PPB_Gas_Reading);
+	eSensorResult = pGasSensor->close();
+	delay_ms(10);
+	return PPB_Gas_Reading;
+}
+
+int32_t ExtractCfgVal()
+{
+	uint8_t buffer[16] = {0};
+	uint8_t read_char;
+	int i = 0;
+	for (i = 0; i < 16; i++) {
+		read_char = GetChar();
+		if (read_char == '\0') break;
+		else buffer[i] = read_char;
+	}
+	DeleteChar(i + 1);
+	return strtol((const char*)buffer, NULL, 10);
+}
+
+uint8_t *int2binString(int32_t a, uint8_t *buff, uint8_t bufSize)
+{
+	buff += (bufSize - 1);
+
+	for (int i = 31; i >= 0; i--) {
+		*buff-- = (a & 1) + '0';
+
+		a >>= 1;
+	}
+
+	return buff;
+}
+
+double Ieee754ConvertToDouble(uint8_t s[32])
```

Comment:
ARM controller typically use LE for representing data inside memory so you could actually just cast the value to double. Something like this float x = *(float *)&Buffer;

Reply:
I will forward this one to Michal and see what he thinks. Some of the other comments were centered around portability. Would soing something like this be an issue for portability also if we aim to port this code?